server/mediaserver.go

```diff
@@ -407,25 +408,36 @@ func jsonProfileToVideoProfile(resp *authWebhookResponse) ([]ffmpeg.VideoProfile
 	return profiles, nil
 }
 
-func jsonDetectionToDetectionConfig(resp *authWebhookResponse) core.DetectionConfig {
-	sceneClassification := ffmpeg.SceneClassificationProfile{
-		SampleRate: resp.Detection.SceneClassificationProfile.SampleRate,
-		Classes:    []ffmpeg.DetectorClass{},
-	}
-	for _, class := range resp.Detection.SceneClassificationProfile.Classes {
-		sceneClassification.Classes = append(sceneClassification.Classes,
-			ffmpeg.DetectorClass{
-				ID:   class.ID,
-				Name: class.Name,
-			})
-	}
-	glog.V(common.DEBUG).Infof("Parsed scene classification config from webhook %v with freq=%v",
-		sceneClassification, resp.Detection.Freq)
+func jsonDetectionToDetectionConfig(resp *authWebhookResponse) (core.DetectionConfig, error) {
 	detection := core.DetectionConfig{
-		Freq:     resp.Detection.Freq,
-		Profiles: []ffmpeg.DetectorProfile{&sceneClassification},
+		Freq:               resp.Detection.Freq,
+		SelectedClassNames: []string{},
+		Profiles:           []ffmpeg.DetectorProfile{},
+	}
+	for _, class := range resp.Detection.SceneClassification {
+		c, ok := ffmpeg.SceneClassificationProfileLookup[class.ClassName]
+		if ok {
+			// Check if a previous selected class had the same model, if yes we can skip the profile
+			skipProfile := false
+			for _, selectedClass := range detection.SelectedClassNames {
+				sc, ok := ffmpeg.SceneClassificationProfileLookup[selectedClass]
+				if ok && sc.ModelPath == c.ModelPath {
+					skipProfile = true
+					break
+				}
+			}
+			detection.SelectedClassNames = append(detection.SelectedClassNames, class.ClassName)
```

Comment:
If we didn't have SelectedClassNames we could still define a local map to track already parsed profiles to determine whether we should skip a profile or not.

Ex.

```golang
var modelPaths map[string]bool
for _, class := range resp.Detection.SceneClassification {
    c, ok := ffmpeg.SceneClassificationProfileLookup[class.Name]
    if !ok {
        ...
    }
    if _, ok := modelPaths[c.ModelPath]; ok {
        // Skip this profile because we already have a profile with a model that covers this class
        continue
    }
    
    c.SampleRate = resp.Detection.SampleRate
    detection.Profiles = append(detection.Profiles, &c)
}
```

I think the nice thing about using a local map rather than SelectedClassNames is that we can avoid carrying additional state in the DetectionConfig struct and we keep everything local to this function since it is not required elsewhere.

Reply:
This is much cleaner even if we decide to stick with `SelectedClassNames` - will fix.